service/src/test/java/uk/nhs/adaptors/gp2gp/ehr/mapper/diagnosticreport/DiagnosticReportMapperTest.java

```diff
@@ -118,15 +133,52 @@ public void tearDown() {
 
     @ParameterizedTest
     @MethodSource("resourceFileParams")
-    public void When_MappingDiagnosticReportJson_Expect_CompoundStatementXmlOutput(String inputJson, String outputXml) {
+    void When_MappingDiagnosticReportJson_Expect_CompoundStatementXmlOutput(String inputJson, String outputXml) {
         final CharSequence expectedOutputMessage = ResourceTestFileUtils.getFileContent(TEST_FILE_DIRECTORY + outputXml);
-        final String jsonInput = ResourceTestFileUtils.getFileContent(TEST_FILE_DIRECTORY + inputJson);
-        final DiagnosticReport diagnosticReport = new FhirParseService().parseResource(jsonInput, DiagnosticReport.class);
+        final DiagnosticReport diagnosticReport = getDiagnosticReportResourceFromJson(inputJson);
 
         final String outputMessage = mapper.mapDiagnosticReportToCompoundStatement(diagnosticReport);
+
         assertThat(removeLineEndings(outputMessage)).isEqualTo(removeLineEndings(expectedOutputMessage.toString()));
     }
 
+    @Test
+    void When_DiagnosticReport_With_NopatMetaSecurity_Expect_ConfidentialityCodeWithinCompoundStatement() {
+        final String testFile = "diagnostic-report-with-multi-specimens-nopat.json";
+        final DiagnosticReport diagnosticReport = getDiagnosticReportResourceFromJson(testFile);
+
+        when(confidentialityService.generateConfidentialityCode(diagnosticReportCaptor.capture()))
+            .thenReturn(Optional.of(NOPAT_CONFIDENTIALITY_CODE));
+
+        final String result = mapper.mapDiagnosticReportToCompoundStatement(diagnosticReport);
+
+        assertAll(
+            () -> assertThat(result).contains(NOPAT_CONFIDENTIALITY_CODE),
+            () -> assertThat(ConfidentialityCodeUtility.getSecurityCodeFromResource(diagnosticReport)).isEqualTo("NOPAT")
+        );
+    }
+
+    @Test
+    void When_DiagnosticReport_With_NoscrubMetaSecurity_Expect_ConfidentialityCodeNotWithinCompoundStatement() {
+        final String testFile = "diagnostic-report-with-multi-specimens-noscrub.json";
+        final DiagnosticReport diagnosticReport = getDiagnosticReportResourceFromJson(testFile);
+
+        when(confidentialityService.generateConfidentialityCode(diagnosticReportCaptor.capture()))
+            .thenReturn(Optional.empty());
```

Comment:
The captor isn't required here, we can just pass the resource itself.

```suggestion
        when(confidentialityService.generateConfidentialityCode(diagnosticReport))
            .thenReturn(Optional.empty());
```